agent/engine/common_integ_testutil.go

```diff
@@ -55,14 +55,14 @@ func init() {
 	sdkClientFactory = sdkclientfactory.NewFactory(context.TODO(), dockerEndpoint)
 }
 
-func defaultTestConfigIntegTest() *config.Config {
+func DefaultTestConfigIntegTest() *config.Config {
```

Comment:
I was wondering if we should put the test utils in a separate `engine_test` package to keep them separate from non-test code in `engine` package? WDYT?

Reply:
For example, `engine.Setup` function doesn't seem like a test function.

Reply:
Unfortunately, this introduces an import cycle. The helper functions use `engine.TaskEngine`, so if they're moved into a separate `engine_test` package, `engine_test` will import `engine`. But `engine` needs to import `engine_test` to access the helper functions in its integration tests.

The import cycle could be broken by moving all the tests out of `engine` and into `engine_test` (`engine` will stop importing `engine_test`), but the tests access a lot of unexported symbols from the `engine` package.

I've renamed some of the ambiguous helper functions to make it clear they're for testing in https://github.com/aws/amazon-ecs-agent/pull/4278/commits/2e3a967a249b7d864002d05d2b0bae173a450016.

Reply:
Thanks!